front/src/applications/operationalStudies/v2/components/ScenarioHeader.tsx

```diff
@@ -1,29 +1,47 @@
-import { Fragment, useCallback, useEffect, useRef, useState } from 'react';
+import { Fragment, useCallback, useEffect, useMemo, useRef, useState } from 'react';
 
-import { X, ChevronDown, Question } from '@osrd-project/ui-icons';
+import { X, ChevronDown, Question, ChevronUp } from '@osrd-project/ui-icons';
 import cx from 'classnames';
 import { useTranslation } from 'react-i18next';
 
 import useAuth from 'utils/hooks/useAuth';
+import type { InfraWithState, ScenarioResponse } from 'common/api/generatedEditoastApi';
+import InfraLoadingState from 'applications/operationalStudies/components/Scenario/InfraLoadingState';
+import { useModal } from 'common/BootstrapSNCF/ModalSNCF';
+import AddAndEditScenarioModal from 'modules/scenario/components/AddOrEditScenarioModal';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
 
 type ScenarioHeaderProps = {
-  scenarioName: string;
+  scenario: ScenarioResponse;
+  infra: InfraWithState;
 };
 
-const ScenarioHeader = ({ scenarioName }: ScenarioHeaderProps) => {
+const ScenarioHeader = ({ scenario, infra }: ScenarioHeaderProps) => {
   const { username } = useAuth();
-  const { t } = useTranslation('common');
+  const { openModal } = useModal();
+
+  const { t } = useTranslation(['translation', 'common', 'operational-studies']);
   const [activeBoards, setActiveBoards] = useState<string[]>([]);
   const [isTruncated, setIsTruncated] = useState({
     scenarioName: false,
     username: false,
   });
+  const [isScenarioDetailsVisible, setIsScenarioDetailsVisible] = useState(false);
 
   const scenarioNameRef = useRef<HTMLSpanElement>(null);
   const usernameRef = useRef<HTMLButtonElement>(null);
 
   const boards = ['Traits', 'Map', 'Macro', 'STD', 'SDD', 'Table', 'Conflicts'];
 
+  const { data: electricalProfileSets } =
+    osrdEditoastApi.endpoints.getElectricalProfileSet.useQuery();
+
+  const electricalProfileSet = useMemo(
+    () =>
+      electricalProfileSets?.find((profile) => profile.id === scenario.electrical_profile_set_id),
+    [electricalProfileSets]
+  );
```

Comment:
nit: since electricalProfileSets is not used anywhere else in the application, you can use RTK selectFromResults:

```suggestion
  const { electricalProfileSet } = osrdEditoastApi.endpoints.getElectricalProfileSet.useQuery(
    undefined,
    {
      selectFromResult: (response) => ({
        ...response,
        electricalProfileSet: response.data?.find(
          (profile) => profile.id === scenario.electrical_profile_set_id
        ),
      }),
    }
  );
```